DataLayer/MpptData/I_MpptData.h

```diff
@@ -7,157 +7,218 @@ class I_MpptData : public QObject
-	virtual double mppt7VoltageOut() const = 0;
-	virtual double mppt7CurrentOut() const = 0;
-	virtual QString mppt7Type() const = 0;
-	virtual QString mppt7Mode() const = 0;
-
-   	/*ArrayData "Sets"*/
-	virtual void setMppt1VoltageIn(double mppt1VoltageIn) = 0;
-	virtual void setMppt1CurrentIn(double mppt1CurrentIn) = 0;
-	virtual void setMppt1VoltageOut(double mppt1VoltageOut) = 0;
-	virtual void setMppt1CurrentOut(double mppt1CurrentOut) = 0;
-	virtual void setMppt1Type(QString mppt1Type) = 0;
-	virtual void setMppt1Mode(QString mppt1Mode) = 0;
-
-	virtual void setMppt2VoltageIn(double mppt2VoltageIn) = 0;
-	virtual void setMppt2CurrentIn(double mppt2CurrentIn) = 0;
-	virtual void setMppt2VoltageOut(double mppt2VoltageOut) = 0;
-	virtual void setMppt2CurrentOut(double mppt2CurrentOut) = 0;
-	virtual void setMppt2Type(QString mppt2Type) = 0;
-	virtual void setMppt2Mode(QString mppt2Mode) = 0;
-
-	virtual void setMppt3VoltageIn(double mppt3VoltageIn) = 0;
-	virtual void setMppt3CurrentIn(double mppt3CurrentIn) = 0;
-	virtual void setMppt3VoltageOut(double mppt3VoltageOut) = 0;
-	virtual void setMppt3CurrentOut(double mppt3CurrentOut) = 0;
-	virtual void setMppt3Type(QString mppt3Type) = 0;
-	virtual void setMppt3Mode(QString mppt3Mode) = 0;
-
-	virtual void setMppt4VoltageIn(double mppt4VoltageIn) = 0;
-	virtual void setMppt4CurrentIn(double mppt4CurrentIn) = 0;
-	virtual void setMppt4VoltageOut(double mppt4VoltageOut) = 0;
-	virtual void setMppt4CurrentOut(double mppt4CurrentOut) = 0;
-	virtual void setMppt4Type(QString mppt4Type) = 0;
-	virtual void setMppt4Mode(QString mppt4Mode) = 0;
-
-	virtual void setMppt5VoltageIn(double mppt5VoltageIn) = 0;
-	virtual void setMppt5CurrentIn(double mppt5CurrentIn) = 0;
-	virtual void setMppt5VoltageOut(double mppt5VoltageOut) = 0;
-	virtual void setMppt5CurrentOut(double mppt5CurrentOut) = 0;
-	virtual void setMppt5Type(QString mppt5Type) = 0;
-	virtual void setMppt5Mode(QString mppt5Mode) = 0;
-
-	virtual void setMppt6VoltageIn(double mppt6VoltageIn) = 0;
-	virtual void setMppt6CurrentIn(double mppt6CurrentIn) = 0;
-	virtual void setMppt6VoltageOut(double mppt6VoltageOut) = 0;
-	virtual void setMppt6CurrentOut(double mppt6CurrentOut) = 0;
-	virtual void setMppt6Type(QString mppt6Type) = 0;
-	virtual void setMppt6Mode(QString mppt6Mode) = 0;
-
-	virtual void setMppt7VoltageIn(double mppt7VoltageIn) = 0;
-	virtual void setMppt7CurrentIn(double mppt7CurrentIn) = 0;
-	virtual void setMppt7VoltageOut(double mppt7VoltageOut) = 0;
-	virtual void setMppt7CurrentOut(double mppt7CurrentOut) = 0;
-	virtual void setMppt7Type(QString mppt7Type) = 0;
-	virtual void setMppt7Mode(QString mppt7Mode) = 0;
+   virtual ~I_MpptData() {}
+
+   /*ArrayData "Gets"*/
+   virtual double mppt1VoltageIn() const = 0;
+   virtual double mppt1CurrentIn() const = 0;
+   virtual double mppt1VoltageOut() const = 0;
+   virtual double mppt1CurrentOut() const = 0;
+   virtual double mppt1PowerIn() const = 0;
```

Comment:
@JordanHeinrichs I recall you mentioning that the new MPPTs don't give both voltage and current so we can't calculate power and efficiency. Is this something that should be reflected in this UI?